### DISCUSSION TITLE
Where can I find a reference on how Photoprism deals with XMP files?

### DISCUSSION THREAD

**Discussion by asitemade4u** [photoprism/photoprism#2331] [Q&A]
Hi all,
I am using Photoprism in conjunction with digiKam which I prefer for geocoding and tagging.
So, my workflow is: camera > darktable (conversion and retouching) > digikam (geocoding and tagging) > PhotoPrism (asset management)
I have asked digiKam to write tags and geolocation only in the XMP file attached to each photo. As a result, PhotoPrism inherits TWO sets of metadata:

1. one in the photo itself, coming mainly from the camera (aperture, speed, etc.)
2. the other in the attached XMP file, coming from digiKam: tags, geolocation, copyright, etc.

My issue is that **PhotoPrism only catches ONE set of metadata** (the first one) and not the metadata coming from the XMP file.

As a sidebar, other asset management tools, such as HomeGallery, do recognize and import both sets of metada.

Please help,
S.

**Comment by lastzero** [comment]
- https://photoprism.app/kb/metadata
- https://docs.photoprism.app/developer-guide/metadata/xmp/

**Comment by asitemade4u** [comment]
Thank you @lastzero for your quick response.
However I find the current behavior of PhotoPrism hard to understand: why not ingest BOTH sources of metadata? There is already a [conversation in Github between you and a user in 2020](https://github.com/photoprism/photoprism/issues/488) and the issue seems still pending.
Please confirm.
BTW I tried to implement `PHOTOPRISM_SIDECAR_JSON` as `true` in my `docker-compose.yml` file, to no avail.

**Reply by lastzero** [reply]
Exiftool will extract embedded XMP Metadata as shown in the tag columns of the document you find at the URL provided to you above.

**Reply by graciousgrey** [reply]
PhotoPrism does read from both sources, see: https://docs.photoprism.app/user-guide/advanced/metadata/
It might be that not all xmp fields you are using are supported yet. 
Feel free to open issues for xmp fields that are not yet supported and attach example xmp files :)

**Reply by asitemade4u** [reply]
Hi @lastzero and @graciousgrey , I attach a sample and fairly typical `XMP` file (which I had to convert because GH does not accept the extension `XMP`).
As you can see:

- I have used darktable upstream to enhance the picture
- digiKam effectively exports tag hierarchies in the XMP file, e.g.:

```
 <digiKam:TagsList>
    <rdf:Seq>
     <rdf:li>GWD</rdf:li>
     <rdf:li>CEM/cem_styles+themes/cem_fence</rdf:li>
     <rdf:li>BRO/Brooklyn</rdf:li>
    </rdf:Seq>
```
Here:

- `CEM` is level 1
- `cem_styles+themes` is level 2
- `cem_fence` is the keyword

It might well be the fact that I am using **hierarchical tags** that does not go well with PP...

[_DSC5362.jpg.txt](https://github.com/photoprism/photoprism/files/8729256/_DSC5362.jpg.txt)

**Reply by graciousgrey** [reply]
Hierarchical tags are not implemented yet but on our to do: https://github.com/photoprism/photoprism/issues/1779.  Once we have support for them we will make sure to read the digikam tags :)

In the meantime: As far as I know digikam can also write the keywords (not levels) to the exif field Keywords. This one is supported by PhotoPrism.

**Reply by graciousgrey** [reply]
Other related issues:
- https://github.com/photoprism/photoprism/issues/2260
- https://github.com/photoprism/photoprism/issues/1828
- https://github.com/photoprism/photoprism/issues/747
- https://github.com/photoprism/photoprism/issues/2075

**Reply by asitemade4u** [reply]
Thank you @graciousgrey for putting it into a longer context. Actually, I have a hunch this kind of questions will come more and more frequently, mostly with people dealing with photos in academic circles or for their work.
The path is logical:

1. TensorFlow is _very_ inefficient when dealing with photos as soon as they are a bit more complicated to read than pet and family photos (I can send you hilarious examples out of my installation of PP if you wish)
3. so, there is a need for a more thorough tagging system
4. PP does not have the tools nor intent to enhance photos so one needs digiKam/LightRoom/darktable to do so.
5. as it happens, each of those tools has good tagging capabilities and uses `XMP` files to manage metadata
6. Consequence: you'd rather have a top-notch metadata importing system in PP.

**Reply by graciousgrey** [reply]
Thank you very much for your feedback! 

We know that there are other needs in professional environments besides object recognition with Tensorflow, that's why the following is planned for the long term:
- Read more metadata fields
- Batch edit 
- Support for hierarchical tags in PhotoPrism
- Write metadata to xmp/exif

**Reply by asitemade4u** [reply]
Great news, looking forward to ALL of it.

**Reply by asitemade4u** [reply]
And yes, there is definitely a need for PP in more professional and academic settings, which in return will enhance your finances ;o))

**Comment by asitemade4u** [comment]
Dear @lastzero , that is not my case as I have sponsored you on GitHub the **day after** I began using PP -- and I even sponsored you, at first, way more than what our capabilities were, but re-sponsored you after being reimbursed of our first mistake ;o)) and had to bother @graciousgrey for it...

S.

**Reply by lastzero** [reply]
Thank you! Was expecting you get it wrong, so this particular reply is already removed. I was trying to explain for everyone why this is not all implement yet, which was your question. In fact no project, company or product has zero open issues. I'm out of the office right now and have to type all answers on my phone, so sorry if this caused any misunderstanding and I could not provide you with a long answer and links to all our related resources. @graciousgrey provided you with more details at a later time. We hope this information is helpful.

**Reply by asitemade4u** [reply]
Of course it is, I did not take any offense...
